go/vt/sqlparser/parse_test.go

```diff
@@ -1034,6 +1034,9 @@ var (
 		input: "alter table a add fulltext index idx (id)",
 	}, {
 		input: "alter table a add foreign key (id) references f (id)",
+	}, {
+		input:  "alter table a add foreign key the_idx (id) references f (id)",
+		output: "alter table a add foreign key (id) references f (id)",
```

Comment:
Should we also change the Format method of foreign key constraint to also print the constraint name if it is passed? If we do that then the expected output would be the same as the input

Reply:
> to also print the constraint name if it is passed?

I considered this, but I think not, because MySQL's `SHOW CREATE TABLE` does not show it.

Reply:
This is the output I get running on MyCLI 
```sql
MySQL root@(none):test> show create table testing2;
+----------+-----------------------------------------------------------------------------------------------------------------------------------+
| Table    | Create Table                                                                                                                      |
+----------+-----------------------------------------------------------------------------------------------------------------------------------+
| testing2 | CREATE TABLE `testing2` (\n  `id` bigint(20) DEFAULT NULL,\n  `fk` bigint(20) DEFAULT NULL\n) ENGINE=InnoDB DEFAULT CHARSET=utf16 |
+----------+-----------------------------------------------------------------------------------------------------------------------------------+
1 row in set
Time: 0.007s
MySQL root@(none):test> alter table testing2 add foreign key new_fk_const (fk) references product112s (id);
You're about to run a destructive command.
Do you want to proceed? (y/n): y
Your call!
Query OK, 0 rows affected
Time: 0.040s
MySQL root@(none):test> show create table testing2;
+----------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table    | Create Table                                                                                                                                                                                                                                        |
+----------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| testing2 | CREATE TABLE `testing2` (\n  `id` bigint(20) DEFAULT NULL,\n  `fk` bigint(20) DEFAULT NULL,\n  KEY `new_fk_const` (`fk`),\n  CONSTRAINT `testing2_ibfk_1` FOREIGN KEY (`fk`) REFERENCES `product112s` (`id`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf16 |
+----------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set
Time: 0.007s
```
there is a key with the name `new_fk_const` which otherwise would be lost if we do not include the key name in the alter table statement. Am I missing something?

Reply:
I think our AST should be as minimally surprising as possible. If a constraint name is parsed, it should also be outputted in `Format`.
If we want to remove the constraint name, we should alter the struct instead of changing the `Format` method.

Reply:
index name added in `Format()`